src/createTRPCJotai.ts

```diff
@@ -69,17 +69,17 @@ const atomWithQuery = <TProcedure extends AnyQueryProcedure, TClient>(
   return queryAtom;
 };
 
-const atomWithMutation = <TProcedure extends AnyMutationProcedure, TClient>(
+const atomWithMutation = <TProcedure extends AnyTRPCMutationProcedure, TClient>(
   path: string[],
   getClient: (get: Getter) => TClient,
 ) => {
-  type Args = ProcedureArgs<TProcedure['_def']>;
+  type Args = inferProcedureInput<TProcedure>;
   type Output = inferProcedureOutput<TProcedure>;
   const mutationAtom = atom(
     null as Output | null,
     async (get, set, args: Args) => {
```

Comment:
Looks good. Let's rename it to `input: Input` from `args: Args` as you suggested.